src/controllers/diff.controller.ts

```diff
@@ -0,0 +1,41 @@
+
+import { NextFunction, Request, Response, Router } from 'express';
+import Ajv from 'ajv';
+
+import { diff } from "@asyncapi/diff";
+
+import { Controller} from '../interfaces';
+
+import { documentValidationMiddleware } from '../middlewares/document-validation.middleware';
+
+
+export class DiffController implements Controller {
+    public basepath = '/diff';
+    private ajv: Ajv;
+
+    private async difference(req: Request, res: Response, next: NextFunction) {
+        try {
+            const output = await diff(req.body.asyncapi, req.body.other);
```

Comment:
There is an important step we should do before calling `diff`, which is to validate the AsyncAPI document that is sent. Like the `diff` docs say: "(...) but you must make sure that the document is valid and dereferenced.", so you can check out how we use `parser` in the `validate` controller 🙂